go test fails ?

- My router has upnp enabled and initially upnp page shows nothing.
- I clone this repo and run "go test":
```
--- FAIL: TestConcurrentUPNP (6.43s)
	upnp_test.go:34: context canceled
	...
	upnp_test.go:34: context canceled
--- FAIL: TestIGD (3.13s)
	upnp_test.go:55: Your external IP is: X.Y.Z.W
	upnp_test.go:60: goupnp: error performing SOAP HTTP request: Post http://192.168.1.1:1900/upnp/control/WANIPConn1: EOF
FAIL
exit status 1
FAIL	github.com/NebulousLabs/go-upnp 9.571s
```
- After that my router's upnp page shows:
```
ID | App Description | External Port | Protocol | Internal Port | IP Address | Status
1 | upnp test | 9001 | TCP | 9001 | 192.168.1.X | Enabled
```
- go1.9.2 linux/amd64

--- a/upnp.go
+++ b/upnp.go
@@ -63,17 +63,23 @@ func (d *IGD) Forward(port uint16, desc string) error {
 		return err
 	}
 
+	time.Sleep(time.Millisecond)
 	err = d.client.AddPortMapping("", port, "TCP", port, ip, true, desc, 0)
 	if err != nil {
 		return err
 	}
+
+	time.Sleep(time.Millisecond)
 	return d.client.AddPortMapping("", port, "UDP", port, ip, true, desc, 0)
 }
 
 // Clear un-forwards a port, removing it from the router's port mapping table.
 func (d *IGD) Clear(port uint16) error {
+	time.Sleep(time.Millisecond)
 	tcpErr := d.client.DeletePortMapping("", port, "TCP")
+	time.Sleep(time.Millisecond)
 	udpErr := d.client.DeletePortMapping("", port, "UDP")
+
 	// only return an error if both deletions failed
 	if tcpErr != nil && udpErr != nil {
 		return tcpErr

--- a/upnp_test.go
+++ b/upnp_test.go
@@ -71,6 +71,7 @@ func TestIGD(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
+	t.Log("Loc:", loc)
 
 	// connect to router directly
 	d, err = Load(loc)

